visualization/app/codeCharta/util/nodeDecorator.ts

```diff
@@ -48,6 +51,29 @@ export class NodeDecorator {
 			})
 	}
 
+	public static doesExclusionResultInEmptyMap(map: CodeMapNode, blacklist: BlacklistItem[]): boolean {
+		let excludedNodes = 0
+		const flattened = ignore()
+		const excluded = ignore()
+
+		for (const item of blacklist) {
+			const path = CodeMapHelper.transformPath(item.path)
+			item.type === BlacklistType.flatten ? flattened.add(path) : excluded.add(path)
+		}
+
+		const tree = hierarchy(map)
+		const desc = tree.descendants()
+
+		desc.map(node => {
+			const path = CodeMapHelper.transformPath(node.data.path)
+			if (excluded.ignores(path) && CodeMapHelper.isLeaf(node.data)) {
+				excludedNodes++
+			}
```

Comment:
Instead of counting the excluded nodes, let's check for any node that is not excluded and stop right there. That way it's significantly less work :-) That way there's also no need to check the tree leaves length.